pkg/generate/app/pipeline.go

```diff
@@ -38,20 +39,20 @@ func NewImagePipeline(from string, image *ImageRef) (*Pipeline, error) {
 	}, nil
 }
 
-// NewBuildPipeline creates a new pipeline with components that are
-// expected to be built
+// NewBuildPipeline creates a new pipeline with components that are expected to
+// be built.
 func NewBuildPipeline(from string, input *ImageRef, outputDocker bool, strategy *BuildStrategyRef, env Environment, source *SourceRef) (*Pipeline, error) {
 	name, ok := NameSuggestions{source, input}.SuggestName()
 	if !ok {
-		name = fmt.Sprintf("app%d", rand.Intn(10000))
+		return nil, ErrNameRequired
 	}
+	name = namer.LimitLength(name, v2.RepositoryNameTotalLengthMax-kutil.DNS1123LabelMaxLength-1)
```

Comment:
Do we know why that line was there and who added it?

On Tue, Oct 13, 2015 at 1:53 PM, Rodolfo Carvalho <notifications@github.com>
wrote:

> In pkg/generate/app/pipeline.go
> <https://github.com/openshift/origin/pull/5082#discussion_r41900739>:
>
> >  	}
> > +	name = namer.LimitLength(name, v2.RepositoryNameTotalLengthMax-kutil.DNS1123LabelMaxLength-1)
>
> In this case here it was output image stream name. I removed that line
> already.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/5082/files#r41900739>.
>